dace/transformation/dataflow/redundant_array.py

```diff
@@ -260,11 +260,25 @@ def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
         e1 = graph.edges_between(in_array, out_array)[0]
         _, b1_subset = _validate_subsets(e1, sdfg.arrays)
 
-        # In strict mode, make sure the memlet covers the removed array
         if strict:
+            # In strict mode, make sure the memlet covers the removed array
             if any(m != a
                    for m, a in zip(b1_subset.size(), out_desc.shape)):
                 return False
+            # In strict mode, abort if the state has two more access nodes for
+            # in array and at least one is a write access. There might be a
+            # RW, WR, or WW dependency.
+            # NOTE: Can we make relax this somehow?
+            accesses = [n for n in graph.nodes()
+                        if isinstance(n, nodes.AccessNode) and
+                        n.desc(sdfg) == in_desc]
+            if len(accesses) > 1:
+                if any(a.access in (
+                        dtypes.AccessType.ReadWrite,
+                        dtypes.AccessType.WriteOnly)
+                       for a in accesses):
```

Comment:
I'd check for in_degree instead, safer

Reply:
Good idea, then it doesn't depend on how it was set by the user or the frontend.